### ISSUE TITLE
[Bug]: <fluent-button/> aria-label attribute cannot be used on a fluent-button with no valid role attribute.

### ISSUE THREAD

**Issue by SSk123** [microsoft/fluentui#25163]
### Library

Web Components (@fluentui/web-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (20) x64 Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz
    Memory: 13.14 GB / 31.80 GB
Browsers:
    Edge: Version 106.0.1370.42 (Official build) (x86_64),
    Chromium Version 105.0.5195.125 (Official Build) (x86_64)
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/team/fluentuiwc/pen/MWvvgNK

### Bug Description

When aria-label or title attribute is passed to <fluent-button/> getting Accessibility fast pass failure on the component.

Passing down role="button" to the fluent-button element seems to fix the issue, this should be added by default on the fluent-button element.

## Actual Behavior
``<fluent-button aria-label="learn more"></fluent-button>``
throws accessibility fast pass failure.

**aria-label attribute cannot be used on a fluent-button with no valid role attribute.**

## Expected Behavior
fluent-button element should have a default role="button" attribute attached to it.

`<fluent-button role="button"></fluent-button>`

### Requested priority

Normal

### Products/sites affected

https://www.msn.com/en-us/health 

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by chrisdholt** [comment]
Thanks for the issue here - First, please ensure you're running these tests with all custom elements in "open" shadow mode. Accessibility insights and other tools have made it clear that they cannot reliably pierce closed shadow DOM. What that means in this case is that the host element is seen, but the element within which actually receives keyboard focus and is read by assistive technology are not. Running in "open" mode will fix this. Please update all references in your project accordingly.

I've taken a look at the link provided and most of the elements in reference (`<fluent-button></fluent-button>`) on the page are indeed running in "closed" mode. Given that, I'm going to close this as I believe running in "open" mode will resolve your fast pass issue.
